### PR TITLE
[DPP-417][DPP-595] Error codes switching - follow-up

### PR DESCRIPTION
**Pull request by pbatko-da**
Renaming CLI flag `--use-error-codes-v2` to `--use-self-service-error-codes`

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by pbatko-da** [comment]
FYI @miklos-da @tudor-da

File: ledger/participant-integration-api/src/main/scala/platform/apiserver/ErrorCodesVersionSwitcher.scala [hunk 1]
```diff
@@ -5,12 +5,18 @@ package com.daml.platform.apiserver
 
 import io.grpc.StatusRuntimeException
 
-final class ErrorCodesVersionSwitcher(enableErrorCodesV2: Boolean) {
+/** A mechanism to switch between the previous error codes (v1) and the new self-service error codes (v2).
```
Comment (tudor-da) [inline ledger/participant-integration-api/src/main/scala/platform/apiserver/ErrorCodesVersionSwitcher.scala:8]:
```suggestion
/** A mechanism to switch between the legacy error codes (v1) and the new self-service error codes (v2).
```

File: ledger/participant-integration-api/src/main/scala/platform/apiserver/ErrorCodesVersionSwitcher.scala [hunk 2]
```diff
@@ -5,12 +5,18 @@ package com.daml.platform.apiserver
...
+  * This class is intended to facilitate transition to self-service error codes.
+  * Once the previous error codes are remvoed and this class should be dropped as well.
```
Comment (tudor-da) [inline ledger/participant-integration-api/src/main/scala/platform/apiserver/ErrorCodesVersionSwitcher.scala:10]:
```suggestion
  * Once the previous error codes are removed, this class should be dropped as well.
```

File: ledger/participant-integration-api/src/main/scala/platform/apiserver/ErrorCodesVersionSwitcher.scala [hunk 3]
```diff
@@ -5,12 +5,18 @@ package com.daml.platform.apiserver
...
+  *
+  * See related epic: https://digitalasset.atlassian.net/browse/DPP-417
```
Comment (tudor-da) [inline ledger/participant-integration-api/src/main/scala/platform/apiserver/ErrorCodesVersionSwitcher.scala:12]: Can you please remove this link?

File: ledger/participant-integration-api/src/main/scala/platform/apiserver/ErrorCodesVersionSwitcher.scala (hunk 1 above)
Comment (pbatko-da) [reply ledger/participant-integration-api/src/main/scala/platform/apiserver/ErrorCodesVersionSwitcher.scala:8]: Ok


I wonder if you could avoid "legacy" 

> legacy
> If possible, use a more precise term. If you do use legacy, include or point to a definition to clarify what you mean in the current context. Do not use legacy with any sort of pejorative connotation.
(https://developers.google.com/style/word-list#letter-l)

File: ledger/participant-integration-api/src/main/scala/platform/apiserver/ErrorCodesVersionSwitcher.scala (hunk 3 above)
Comment (pbatko-da) [reply ledger/participant-integration-api/src/main/scala/platform/apiserver/ErrorCodesVersionSwitcher.scala]: Ok

File: ledger/participant-integration-api/src/main/scala/platform/apiserver/ErrorCodesVersionSwitcher.scala (hunk 2 above)
Comment (pbatko-da) [reply ledger/participant-integration-api/src/main/scala/platform/apiserver/ErrorCodesVersionSwitcher.scala:10]: Ok

**Review by tudor-da — APPROVED** [review]

File: ledger/participant-integration-api/src/main/scala/platform/apiserver/ErrorCodesVersionSwitcher.scala (hunk 1 above)
Comment (tudor-da) [reply ledger/participant-integration-api/src/main/scala/platform/apiserver/ErrorCodesVersionSwitcher.scala]: Huh, I didn't know it has potentially negative connotation

**Review by miklos-da — APPROVED** [review]
Thanks for following up on this, @pbatko-da!

**Pull request merged**